runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6372,7 +6372,202 @@ TR_J9ByteCodeIlGenerator::loadFromCP(TR::DataType type, int32_t cpIndex)
+                     autoboxClassSig = "Ljava/lang/Float;";
+                     autoboxClassSigLength = 17;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "F", 1);
+                     break;
+                  case 'D':
+                     autoboxClassSig = "Ljava/lang/Double;";
+                     autoboxClassSigLength = 18;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "D", 1);
+                     break;
+                  case 'B':
+                     autoboxClassSig = "Ljava/lang/Byte;";
+                     autoboxClassSigLength = 16;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "B", 1);
+                     break;
+                  case 'C':
+                     autoboxClassSig = "Ljava/lang/Character;";
+                     autoboxClassSigLength = 21;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "C", 1);
+                     break;
+                  case 'S':
+                     autoboxClassSig = "Ljava/lang/Short;";
+                     autoboxClassSigLength = 17;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "S", 1);
+                     break;
+                  case 'Z':
+                     autoboxClassSig = "Ljava/lang/Boolean;";
+                     autoboxClassSigLength = 19;
+                     typeClassBlock = comp()->fej9()->getClassFromSignature(autoboxClassSig, autoboxClassSigLength, method());
+                     valueOffset = comp()->fej9()->getObjectHeaderSizeInBytes()
+                                          + comp()->fej9()->getInstanceFieldOffset(typeClassBlock, "value", 5, "Z", 1);
+                     break;
+                  default:
```

Comment:
shouldn't this assert?

Reply:
Yes, added a fatal assertion for unrecognized type signature.